core/src/main/java/com/linecorp/armeria/server/http/healthcheck/ManagedHttpHealthCheckService.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.healthcheck;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import com.google.common.base.Ascii;
+
+import com.linecorp.armeria.common.http.AggregatedHttpMessage;
+import com.linecorp.armeria.common.http.HttpData;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * A {@link HttpHealthCheckService} which allows overriding its status via a POST request.
+ * <p>
+ * <h2>Example</h2>
+ * <pre>{@code
+ * Server server = new ServerBuilder().serviceUnder("/health", new ManagedHttpHealthCheckService())
+ *                                    .build()
+ * }</pre>
+ * Default config uses content on or off
+ * You can also use your own path matching.
+ * </p>
+ * <pre>{@code
+ * ...
+ * Server server = new ServerBuilder().serviceUnder("health", new ManagedHttpHealthCheckService() {
+ *                         public CompletableFuture<Optional<Boolean>> mode(HttpRequest req) {
+ *                                 return CompletableFuture.completedFuture(Optional.empty());
+ *                         }
+ *                 }).build())
+ * ...
+ * }</pre>
+ */
+public class ManagedHttpHealthCheckService extends HttpHealthCheckService {
+    private static final AggregatedHttpMessage TURN_ON_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, HttpData.ofAscii("turn on"));
```

Comment:
- `"Set healthy."`
- Could you also set the content type? e.g. `MediaType.PLAIN_TEXT_UTF_8` (You'll have to use `ofUtf8`)